docs/channel/README.md

```diff
@@ -0,0 +1,224 @@
+| Offset repositioning    | Not supported                                           | Not supported                                           | Supported                                                                            |
+| Dataplane stack         | Java (Vert.x)                                           | Go (Sarama)                                             | Go (Sarama)                                                                          |
+
+** Work in progress
+
+## Migration
+
+Automated migration is possible for migrations from the consolidated channel implementation to the new implementation.
+Distributed channel users need to migrate manually.
+
+### Breaking changes from consolidated channel
+
+#### Channel config
+
+New implementation uses a new configmap called `kafka-channel-config`. It has a different structure as well.
+
+The consolidated channel config looks like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-kafka
+  namespace: knative-eventing
+data:
+  version: 1.0.0
+  sarama: |
+    enableLogging: false
+    config: |
+      Version: 2.0.0 # Kafka Version Compatibility From Sarama's Supported List (Major.Minor.Patch)
+      Admin:
+        Timeout: 10000000000  # 10 seconds
+      Net:
+        KeepAlive: 30000000000  # 30 seconds
+        MaxOpenRequests: 1 # Set to 1 for use with Idempotent Producer
+        TLS:
+          Enable: true
+        SASL:
+          Enable: true
+          Version: 1
+      Metadata:
+        RefreshFrequency: 300000000000  # 5 minutes
+      Consumer:
+        Offsets:
+          AutoCommit:
+            Interval: 5000000000  # 5 seconds
+          Retention: 604800000000000  # 1 week
+      Producer:
+        Idempotent: true  # Must be false for Azure EventHubs
+        RequiredAcks: -1  # -1 = WaitForAll, Most stringent option for "at-least-once" delivery.
+  eventing-kafka: |
+    cloudevents:
+      maxIdleConns: 1000
+      maxIdleConnsPerHost: 100
+    kafka:
+      authSecretName: kafka-cluster
+      authSecretNamespace: knative-eventing
+      brokers: my-cluster-kafka-bootstrap.kafka:9092
+    channel:
+      adminType: kafka # One of "kafka", "azure", "custom"
+      dispatcher:
+        cpuRequest: 100m
+        memoryRequest: 50Mi
+      receiver:
+        cpuRequest: 100m
+        memoryRequest: 50Mi
+```
+
+A sample config for the new channel is available below:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kafka-channel-config
+  namespace: knative-eventing
+data:
+  bootstrap.servers: "my-cluster-kafka-bootstrap.kafka:9092"
+  auth.secret.ref.name: kafka-cluster
+  auth.secret.ref.namespace: knative-eventing
+```
+
+Most of the options are not available in the new channel implementation because we are not using Sarama in the new dataplane
```

Comment:
wanna mention this: https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/data-plane/config/channel/100-config-kafka-channel-data-plane.yaml#L20  ?

Reply:
I did now 2 lines below: https://github.com/knative-sandbox/eventing-kafka-broker/pull/2246/files#diff-d45b27d995a6f32b461e07b92174c330d202f9bfd75db4e11970b224f8e7e171R260-R261